### DISCUSSION TITLE
How do I assign single gpu for text generation?

### DISCUSSION THREAD

**Discussion by Kepler-16b** [oobabooga/text-generation-webui#983] [Q&A]
Context: As title says basically. I have two gpu in my system. Both nvdia, one geforce 1080Ti (11gb) and one tesla P40 (24gb). I want 1080Ti to run text generation while P40 handles stable diffusion image generation. 
Question: How can I assign only one gpu in oobabooga web ui? I have checked out the wiki and basic setting guide in repo, and currently it seems I can only use both which I think will be overkill. I only use text generation for prompting, and 11gb vram should be fine.

**Comment by bmoconno** [comment]
>Maxmimum GPU memory in GiB to be allocated per GPU. Example: --gpu-memory 10 for a single GPU, --gpu-memory 10 5 for two GPUs. You can also set values in MiB like --gpu-memory 3500MiB.

I don't have multiple GPUs, so I can't test, but I wonder if you could just set the mem usage on one of the GPUs to 0?

```
--gpu-memory 0 11
```
or 
```
--gpu-memory 11 0
```
depending on the order in which your gpus are recognized.

**Comment by tensiondriven** [comment]
I might be off (way off, like GPT-3 off) on this; but my understanding is:
- P40 has good int8 (8-bit) performance but poor FP16 (16-bit).  This makes it good for text generation using 8-bit mode.  
- Stable Diffusion usually uses 16-bit mode, so the performance of SD on a P40 will suffer greatly compared to another card that does real 16-bit.

I don't know how 4-bit plays into this.  I think that 4-bit needs to be expanded at some point, but I'm not sure which/where.
